### ISSUE TITLE
Compilation on Power fails with clang due to vec_xl

### ISSUE THREAD

**Issue by Flamefire** [pytorch/pytorch#41920]
## 🐛 Bug

When compiling PyTorch with Clang on Power the compilation fails with

```
../aten/src/TH/vector/VSX.cpp:130:24: error: no matching function for call to 'vec_xl'
        x11_fp64vec2 = vec_xl(0, x+(i+22))
```

The reason is clear by inspection of clangs `altivec.h`: Their `vec_xl` function take non-const pointers as their 2nd argument. See e.g. https://github.com/llvm-mirror/clang/blob/master/lib/Headers/altivec.h

I opened a bug report with llvm but don't expect it to be fixed soon: https://bugs.llvm.org/show_bug.cgi?id=46817

Possible solution to workaround is to conditionally (on clang only) add code like this (untested):

```
template<typename T>
auto vec_xl_wrapped(signed long offset, const T* ptr){
  return vec_xl(offset, (T*) ptr);
}
#define vec_xl vec_xl_wrapped
```

So basically introduce a wrapper function which removes the const. Also note that the above requires C++14 for the auto return type. For C++11 a trailing return type is required. This would also work on GCC as `T*` implicitly converts to `const T*` IIRC
